editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/ArgumentStyleChildTableControl.java

```diff
@@ -0,0 +1,362 @@
+    }
+
+    /**
+     * Add a change listener.
+     * 
+     * @param listener new listener
+     */
+    public void addChangeListener(ChangeListener listener) {
+        this.changeListeners.add(listener);
+    }
+
+    /**
+     * Remove a change listener.
+     * 
+     * @param listener old listener
+     */
+    public void removeChangeListener(ChangeListener listener) {
+        this.changeListeners.remove(listener);
+    }
+
+    /**
+     * Update button state based on what's selected.
+     */
+    public void updatePropertyTypeButtons() {
+        if (isReadOnly) {
+            this.addButton.setEnabled(false);
+            this.removeButton.setEnabled(false);
+
+        } else {
+            this.addButton.setEnabled(true);
+            if (getStructuredSelection() != null && !getStructuredSelection().isEmpty()) {
+                removeButton.setEnabled(true);
+            }
+        }
+    }
+
+    /**
+     * @return the current selection from the table
+     */
+    public IStructuredSelection getStructuredSelection() {
+        if (propertyTreeTable != null && !propertyTreeTable.getSelection().isEmpty()) {
+            return (IStructuredSelection) propertyTreeTable.getSelection();
+        }
+        return null;
+    }
+
+    /**
+     * @return warning string
+     */
+    public String getWarning() {
+        return this.warningMsg;
+    }
+
+    protected void removePropertyFromList() {
+    	if (!getStructuredSelection().isEmpty()) {
+    		AbstractCamelModelElement selectedProperty = (AbstractCamelModelElement) getStructuredSelection().getFirstElement();
+    		argumentList.remove(selectedProperty);
+    	}
+    }
+    
+    protected void addPropertyTypeToList() {
+        final ArgumentInputDialog dialog = new ArgumentInputDialog(Display.getCurrent()
+                .getActiveShell());
+        int rtnValue = dialog.open();
+        if (rtnValue == PropertyInputDialog.OK) {
+            final String type = dialog.getArgumentType();
+            final String value = dialog.getArgumentValue();
+            addBeanArgument(type, value);
+        }
+    }
+    
+    public void setInput(Object input) {
+    	if (input instanceof AbstractCamelModelElement) {
+    		inputElement = (AbstractCamelModelElement) input;
+    	}
+    }
+
+    private void addBeanArgument (String type, String value) {
+		String prefixNS = null;
+		boolean isBlueprint = false;
+		AbstractCamelModelElement newRoot = inputElement;
+		if (inputElement instanceof CamelFile && !inputElement.getChildElements().isEmpty() ) {
+			newRoot = inputElement.getChildElements().get(0);
+			prefixNS = newRoot.getRouteContainer().getXmlNode().getPrefix();
+			isBlueprint = ((CamelFile)inputElement).isBlueprint();
+		} else if (inputElement.getRouteContainer() != null) {
+			prefixNS = inputElement.getRouteContainer().getXmlNode().getPrefix();
+		}
+		String tagName = "argument"; //$NON-NLS-1$
+		if (!isBlueprint) {
+			tagName = "constructor-arg"; //$NON-NLS-1$
```

Comment:
I prefer to have the else written. What do you think of?

String tagName;
if(isBluePrint){
tagName = ""argument";
} else {
tagName = "constructor-arg";
}